src/org/opensolaris/opengrok/index/Indexer.java

```diff
@@ -101,542 +109,658 @@ private static void A_usage() {
+            });
+
+            parser.on("-K", "--listRepos", "List all repository paths and exit.").Do( v -> {
+                listRepos = true;
+            });
+
+            parser.on("-k", "--deleteHistory", "=/path/to/repository", 
+                "Delete the history cache for the given repository and exit.",
+                "Use '*' to delete the cache for all repositories.").Do( repo -> {
+                zapCache.add((String)repo);
+            });
+
+            parser.on("-L", "--style", "=path", 
+                "Path to the subdirectory in the web-application containing the",
+                "requested stylesheet. The factory-setting is: \"default\"").
+                Do( stylePath -> {
+                    cfg.setWebappLAF((String)stylePath);
                 }
-
-                // If the webapp is running with a config that does not contain
-                // 'projectsEnabled' property (case of upgrade or transition
-                // from project-less config to one with projects), set the property
-                // using a message so that the 'project/indexed' messages
-                // emitted during indexing do not cause validation error.
-                if (addProjects && host != null && port > 0) {
-                    Message m = Message.createMessage("config");
-                    m.addTag("set");
-                    m.setText("projectsEnabled = true");
-                    m.write(host, port);
+            );
+
+            parser.on("-l", "--lock", "=on/off", onOff, Boolean.class,
+                "Turn on/off locking of the Lucene database during index generation.").Do( v -> {
+                cfg.setUsingLuceneLocking((Boolean)v);
+            });
+
+            parser.on("-m", "--memory", "=number", Double.class,
+                "Amount of memory that may be used for buffering added documents",
+                "and deletions before they are flushed to the directory (default "+Configuration.defaultRamBufferSize+"MB).",
+                "Please increase JVM heap accordingly, too.").Do( memSize -> {
+                cfg.setRamBufferSize((Double)memSize);
+            });
+
+            parser.on("--man", "Generate OpenGrok XML manual page.").Do( v -> {
+                try {
+                    System.out.print(parser.getManPage());
+                } catch(IOException e) {
+                    System.err.println(e.getMessage());
+                    status = 1;
                 }
-
-                // Get history first.
-                getInstance().prepareIndexer(env, searchRepositories, addProjects,
-                        defaultProjects,
-                        listFiles, createDict, subFiles, repositories,
-                        zapCache, listRepos);
-                if (listRepos || !zapCache.isEmpty()) {
-                    return;
+                System.exit(status);
+            });
+
+            parser.on("-N", "--symlink", "=/path/to/symlink", 
```

Comment:
ditto

Reply:
You do realize that removal of these single character options will be breaking what exists in the field?